ci/pipeline/ray_release_test.py

```diff
@@ -0,0 +1,34 @@
+import os
+import subprocess
+import sys
+from ray_logger import get_logger
+from typing import List
+
+
+def main() -> int:
+    """
+    This script determines and runs the right tests for the PR changes. To be used in
+    CI and buildkite environment only.
+    """
+    logger = get_logger()
+    changed_files = _get_changed_files()
+    logger.info(f"Changed files: {changed_files}")
+    return 0
+
+
+def _get_changed_files() -> List[str]:
+    """
+    Get the list of changed files in the current PR.
+    """
+    base_branch = os.environ["BUILDKITE_PULL_REQUEST_BASE_BRANCH"]
```

Comment:
check and throw an exception when `BUILDKITE_PULL_REQUEST_BASE_BRANCH` is empty? like on `occ-ci-build-branch`, `BUILDKITE_PULL_REQUEST_BASE_BRANCH` is always empty I think.

also I think you are looking for is to do what github is doing, which is:

1. take the commit head commit (maybe just use `BUILDKITE_COMMIT` ?)
2. use git-merge-base https://git-scm.com/docs/git-merge-base to find a common ancestor between head and `BUILDKITE_PULL_REQUEST_BASE_BRANCH`
3. diff against the common ancestor with head commit for changes.

Reply:
Sounddd good. Yes, I'm doing 1 and 3 basically, since the base_branch and head is already on the same branch so 2 will return base_branch; also I'm using HEAD for BUILDKITE_COMMIT